### PR TITLE
[GPU] Add unit tests for activation functions for blocked layouts

### PR DESCRIPTION
**Pull request by AlexKramskoy**
### Details:
 - Add unit tests for blocked layouts for GELU-7 operation

### CODE REVIEW

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp [hunk 1]
```diff
@@ -1630,14 +1630,42 @@ TEST_P(activation_random_test, random) {
...
+        {data_types::i8, format::b_fs_yx_fsv32, {1, 32, 5, 5}, activation_func::relu, {}, {}},
+        {data_types::i8, format::bs_fs_yx_bsv32_fsv32, {32, 32, 5, 5}, activation_func::relu, {}, {}},
+        {data_types::f16, format::bs_fs_yx_bsv32_fsv16, {32, 32, 5, 5}, activation_func::relu, {}, {}},
+        {data_types::i8, format::bs_fs_yx_bsv32_fsv32, {16, 16, 5, 5}, activation_func::relu, {}, {}},
+        {data_types::f16, format::bs_fs_yx_bsv32_fsv16, {16, 16, 5, 5}, activation_func::relu, {}, {}},
+        // gelu-7
+        {data_types::f16, format::bfyx, {16, 16, 5, 5}, activation_func::gelu, {}, {}},
+        {data_types::f16, format::bfyx, {16, 16, 5, 5}, activation_func::gelu_tanh, {}, {}},
+        {data_types::f32, format::bfyx, {16, 16, 5, 5}, activation_func::gelu, {}, {}},
+        {data_types::f32, format::bfyx, {16, 16, 5, 5}, activation_func::gelu_tanh, {}, {}},
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1646]: can't you use combine inside? 
Like in example [here](https://github.com/openvinotoolkit/openvino/blob/a544fc70d189fa4b1fed85ebb08bf584ae5649de/src/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/non_max_suppression.cpp#L30)?

Comment (AlexKramskoy) [reply src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp]: Done

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp [hunk 2]
```diff
@@ -1630,14 +1637,38 @@ TEST_P(activation_random_test, random) {
...
+
+INSTANTIATE_TEST_SUITE_P(relu_activation_blocked_tests, activation_random_test, reluParams);
+
+const std::vector<data_types> dataTypes = {data_types::f16, data_types::f32};
+const std::vector<format::type> types = {format::bfyx,
+                                         format::bfzyx,
+                                         format::yxfb,
+                                         format::byxf,
+                                         format::fyxb,
+                                         format::b_fs_yx_fsv2,
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1656]: could you rename 'types' to 'layoutFormats'
Also it's not clear where are
bs_fs_yx_bsv32_fsv32
bs_fs_yx_bsv32_fsv16...

and why do you use b_fs_yx_fsv2 instead?

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp [hunk 3]
```diff
@@ -1630,14 +1637,38 @@ TEST_P(activation_random_test, random) {
...
+                                         format::b_fs_zyx_fsv2};
+const std::vector<activation_func> activationFunctions = {activation_func::gelu,
+                                                          activation_func::gelu_tanh,
+                                                          activation_func::softsign};
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1660]: can we add relu here also? woulld all the tests passed?

**Comment by tgubanova-lohika** [comment]
could you change PR description to mention SoftSign here also (and potentially relu also)

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp (hunk 2 above)
Comment (AlexKramskoy) [reply src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1656]: Added layouts

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp (hunk 3 above)
Comment (AlexKramskoy) [reply src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1707]: Done

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp [hunk 4]
```diff
@@ -1630,14 +1637,86 @@ TEST_P(activation_random_test, random) {
...
-                        ));
+const auto reluParams = testing::ValuesIn(std::vector<activation_random_test_params>{
+    {data_types::i8, format::b_fs_yx_fsv32, {1, 32, 5, 5}, activation_func::relu, {}, {}},
+    {data_types::i8, format::bs_fs_yx_bsv32_fsv32, {32, 32, 5, 5}, activation_func::relu, {}, {}},
+    {data_types::f16, format::bs_fs_yx_bsv32_fsv16, {32, 32, 5, 5}, activation_func::relu, {}, {}},
+    {data_types::i8, format::bs_fs_yx_bsv32_fsv32, {16, 16, 5, 5}, activation_func::relu, {}, {}},
+    {data_types::f16, format::bs_fs_yx_bsv32_fsv16, {16, 16, 5, 5}, activation_func::relu, {}, {}},
+});
+
+INSTANTIATE_TEST_SUITE_P(relu_activation_blocked_tests, activation_random_test, reluParams);
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1648]: what sense to have relu test separately if you already included it to common list?

File: src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp [hunk 5]
```diff
@@ -1630,14 +1637,86 @@ TEST_P(activation_random_test, random) {
...
+                                                          activation_func::selu,
+                                                          activation_func::sign,
+                                                          activation_func::softplus,
+                                                          activation_func::swish,
+                                                          activation_func::hswish,
+                                                          activation_func::mish,
+                                                          activation_func::round_half_to_even,
+                                                          activation_func::round_half_away_from_zero,
+                                                          activation_func::gelu_tanh,
+                                                          activation_func::softsign};
```
Comment (tgubanova-lohika) [inline src/plugins/intel_gpu/tests/test_cases/activation_simple_gpu_test.cpp:1707]: @wilson-seok @yeonbok as we discussed we added all the activation functions here. Commented functions are failed now so as we agreed I propose to create separate tasks for them.

**Review by wilson-seok — APPROVED** [review]

**Review by wilson-seok — APPROVED** [review]

**Review by yeonbok — APPROVED** [review]

**Pull request merged**